sqlx-macros/src/lib.rs

```diff
@@ -60,6 +60,17 @@ macro_rules! async_macro (
         let res: Result<proc_macro2::TokenStream> = block_on(async {
             use sqlx::connection::Connect;
 
+            // If a .env file exists at CARGO_MANIFEST_DIR, load environment variables from this,
+            // otherwise fallback to default dotenv behaviour.
+            if let Ok(env_path) = std::env::var("CARGO_MANIFEST_DIR")
+                .map(|dir| std::path::PathBuf::from(dir).join(".env"))
```

Comment:
Nit: I'd just do `let env_path = Path::from(&dir).join(".env")` in the block. It's easier to read that way.